pkg/reconciler/kubernetes/tektonpipeline/tektonpipeline.go

```diff
@@ -411,3 +417,26 @@ func (m *Recorder) logMetrics(status, version string, logger *zap.SugaredLogger)
 		logger.Warnf("Failed to log the metrics : %v", err)
 	}
 }
+
+func (r *Reconciler) markUpgrade(ctx context.Context, tp *v1alpha1.TektonPipeline) error {
+	labels := tp.GetLabels()
+	ver, ok := labels[tektoninstallerset.ReleaseVersionKey]
+	if ok && ver == r.operatorVersion {
+		return nil
+	}
+	if ok && ver != r.operatorVersion {
+		tp.Status.MarkInstallerSetNotReady("Upgrade Pending")
+		tp.Status.MarkNotReady("Upgrade Pending")
+	}
```

Comment:
```suggestion
	if ok {
	    if ver == r.operatorVersion {
		return nil
	    }
	    tp.Status.MarkInstallerSetNotReady("Upgrade Pending")
	    tp.Status.MarkNotReady("Upgrade Pending")
	}
```

Reply:
i shall make this update. :+1: 
however, i find nested `if` less effective in conveying the logic to the reader generally. :technologist:. ie, reader's mind has to backtrack.